src/service/models.py

```diff
@@ -102,3 +102,32 @@ class HealthResponse(BaseModel):
     """Health check response model."""
 
     status: Annotated[str, Field(description="Health status")]
+
+
+class DeploymentStatus(BaseModel):
+    """Status information of a Kubernetes deployment."""
+
+    available_replicas: Annotated[
+        int, Field(description="Number of available replicas")
+    ] = 0
```

Comment:
I don't think I completely understand what a replica is in this context - masters don't have replicas do they?

Reply:
master doesn't. But I want to make it compatible. Both the Master and Worker components have their own deployments (just 1 deployment per component). The Master deployment always runs a single replica, while the Worker deployment runs multiple replicas (worker nodes).

Reply:
:+1: